homeassistant/components/firetv/media_player.py

```diff
@@ -62,7 +64,8 @@ def has_adb_files(value):
     vol.Optional(CONF_ADB_SERVER_IP): cv.string,
     vol.Optional(
         CONF_ADB_SERVER_PORT, default=DEFAULT_ADB_SERVER_PORT): cv.port,
-    vol.Optional(CONF_GET_SOURCES, default=DEFAULT_GET_SOURCES): cv.boolean
+    vol.Optional(CONF_GET_SOURCES, default=DEFAULT_GET_SOURCES): cv.boolean,
+    vol.Optional(CONF_APPS, default=DEFAULT_APPS): dict,
```

Comment:
Since we know the type of the keys and the values here, we should add validation for them.

Reply:
Like this?

```python
vol.Optional(CONF_APPS, default=DEFAULT_APPS): vol.Schema({cv.string: cv.string})
```

Can't all yaml keys and values be interpreted as strings? So I think this would prevent something like:

```yaml
...
  apps:
    bad1:
      key: value
    bad2:
      - list item
```

Reply:
I changed this to what I posted above, and it works. If that's not what you had in mind, let me know and I'll fix it.